scripts/generate_typos_config.py

```
@@ -0,0 +1,261 @@
+        )
+    )
+    # Two overlapping stems (or a stem colliding with an accepted word) would
+    # emit the same key twice, which typos silently resolves to the last one
+    # and hides a curation mistake. Fail loudly instead so a bad STEMS edit is
+    # caught at generation time rather than in a confusing gate result. The
+    # header is lines[0]; every other line is a `key = "value"` entry.
+    seen: set[str] = set()
+    for line in lines[1:]:
+        key = line.split(' = "', 1)[0]
+        if key in seen:
+            raise ValueError(f"duplicate typos key generated: {key!r}")
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)